posts/2021-08-30-binding-app-to-postgresql-db-with-odo-and-ol-stack.adoc

```diff
@@ -0,0 +1,284 @@
+---
+= Bind an Operator Managed PostgreSQL Database to a JPA Application in a Kubernetes Cluster
+Edward Mezarina <https://github.com/mezarin>
+
+Binding your Microservice application to operator based services (e.g. databases, etc.) has now been made easier through automatic collection of service data, deployment of needed resources, and data sharing with the application.
+
+== Open Liberty devfile stack
+
+The link:https://github.com/OpenLiberty/application-stack#open-liberty-application-stack[Open Liberty devfile stack] provides much of the infrastructure (Open Liberty, Maven/Gradle, Open J9, etc.) needed to start developing applications that use Maven or Gradle, and it is made available as Maven and Gradle development images. These images are used by the devfiles provided by the stack to serve as the base for building and running your applications. 
+
+The Open Liberty devfile stack provides two fully configured link:https://docs.devfile.io/devfile/2.1.0/user-guide/index.html[devfiles]: A link:https://github.com/devfile/registry/blob/main/stacks/java-openliberty/devfile.yaml[Maven based devfile] and a link:https://github.com/devfile/registry/blob/main/stacks/java-openliberty-gradle/devfile.yaml[Gradle based devfile]. These devfiles will allow you to deploy your Maven or Gradle built applications with ease as these devfiles define the environment and set of steps needed to build and deploy your application using the Open Liberty runtime.
+
+== OpenShift Do (odo)
+
+link:https://odo.dev[Odo] is a simple CLI tool that you use to create devfile based components that interact directly with your Kubernetes cluster to setup the environment, build/deploy/access/debug your application, and link services to odo components. Directives to manage the environment and application are provided by a component's devfile.
+
+== Service Binding Operator
+
+The link:https://github.com/redhat-developer/service-binding-operator/blob/master/README.md[Service Binding Operator] makes it easier for developers to bind operator managed services (i.e. databases) to applications. More specifically, the operator will collect service data, deploy resources (i.e. secrets, configmaps) containing that data, and will make the data available to the application through the ServiceBinding custom resource it provides.
+
+== Try it out
+
+To begin, you need three things:
+
+- Kubernetes cluster. For this blog, an OpenShift cluster is used. Be sure to log in. Odo will interact with your kubernetes cluster.
+- OpenShift Do. If you have not already installed odo, do so now by following the instructions outlined in the link:https://odo.dev[odo] documentation. Be sure to install version 2.2.4 and above.
+- An application that uses a PostgreSQL database. For this blog, the Open Liberty stack link:https://github.com/OpenLiberty/application-stack-samples/tree/main/jpa[sample JPA application] is used.
```

Comment:
suggested edit:

For this post, we'll use the Open Liberty stack link:https://github.com/OpenLiberty/application-stack-samples/tree/main/jpa[sample JPA application].